src/services/GasEstimator.ts

```diff
@@ -0,0 +1,124 @@
+import { BlockWithTransactions, FeeData, TransactionResponse } from '@ethersproject/abstract-provider';
```

Comment:
let me compare to original.... need few minutes

Reply:
Just a side note: since TSConfig is different for Sanctuary and canvas, I had to change a few `undefined` values and `Math` operations.
The only test that didn't work was an `NaN` in `.printable`, because of the reason above.
Everything will work on the current implementation.